services.py

```diff
@@ -85,45 +91,83 @@ def get_param_from_config(self, config_path: str, param_name: str):
-        print(address)
-
-    # latitude, longitude = get_current_location()
-    # if latitude and longitude:
-    #     city = get_city_from_coordinates(latitude, longitude)
-    #     if city:
-    #         print(f"The current city is: {city}")
-    #     else:
-    #         print("Could not determine the city.")
-    # else:
-    #     print("Could not determine the current location.")
+        try:
+                gps = serial.Serial(SERIAL_PORT, baudrate = 9600, timeout = 0.5)
+                
+                data = gps.readline().decode('utf-8').rstrip()
+                message = data[0:6]
+                # print(message)
+                if (message == "$GNRMC"):
+                    # GPRMC = Рекомендуемые минимальные конкретные данные GPS / транзит
+                    # Чтение фиксированных данных GPS является альтернативным подходом, 
+                    # который также работает
+                    parts = data.split(",")
+                    print(parts)
+                    if parts[2] == 'V':
+                        # V = Предупреждение, скорее всего, спутников нет в поле зрения ...
+                        print ("GPS receiver warning")
+                    else:
+                        # Получить данные о местоположении, которые были переданы с сообщением GPRMC. 
+                        # В этом примере интересуют только долгота и широта. Для других значений можно
+                        # обратиться к http://aprs.gids.nl/nmea/#rmc
+                        longitude = self.formatDegreesMinutes(parts[5], 3)
+                        latitude = self.formatDegreesMinutes(parts[3], 2)
+                        logger.info("Device position: lon = " + str(longitude) + ", lat = " + str(latitude))
+                else:
+                    # Обрабатывать другие сообщения NMEA и неподдерживаемые строки
+                    pass
+
+                gps.close()
+                return [latitude, longitude]
+        except Exception as e:
+            logger.error(f'Error getting GPS coordinates through serial port: {e}\n Falling back to IP geolocation...')
+            g = geocoder.ip('me')
+            return g.latlng
+    # TODO: Check if delete_file is needed
+    # def delete_file(self, path):
+    #     if os.path.exists(path):
+    #         try:
+    #             os.remove(path)
+    #             logger.info(f'Deleted file: {path}')
+    #         except PermissionError:
+    #             logger.error(f'Error deleting file: {path}')
+
+
+
+    def save_json(self, dictionary, path: str):
+        with open(path, "w") as outfile:
+            json.dump(dictionary, outfile, indent=4)
+
+    def get_current_city(self):
+
+        geolocator = Photon(user_agent="measurements")
+
+        location_lat_lon = self.get_lat_lon()
+
+        Latitude = f"{location_lat_lon[0]}"
+        Longitude = f"{location_lat_lon[1]}"
+        
+        location = geolocator.reverse(Latitude+","+Longitude)
```

Comment:
Usually it is written this way
```suggestion
        location = geolocator.reverse(f"{Latitude},{Longitude}")
```

Reply:
👍I found a StackOverflow question that used the fllowing syntax, therefore I used it also.